src/data/polynomial/algebra_map.lean

```diff
@@ -97,17 +97,15 @@ end comm_semiring
 section aeval
 variables [comm_semiring R] {p : polynomial R}
 
--- TODO this could be generalized: there's no need for `A` to be commutative,
--- we just need that `x` is central.
-variables [comm_semiring A] [algebra R A]
-variables {B : Type*} [comm_semiring B] [algebra R B]
+variables [semiring A] [algebra R A]
+variables {B : Type*} [semiring B] [algebra R B]
 variables (x : A)
 
 /-- Given a valuation `x` of the variable in an `R`-algebra `A`, `aeval R A x` is
 the unique `R`-algebra homomorphism from `R[X]` to `A` sending `X` to `x`. -/
 def aeval : polynomial R →ₐ[R] A :=
 { commutes' := λ r, eval₂_C _ _,
-  ..eval₂_ring_hom (algebra_map R A) x }
+  ..eval₂_ring_hom_noncomm (algebra_map R A) (λ a b, (algebra.commutes a b).symm) x }
```

Comment:
I guess this must have been introduced in an earlier PR that I missed, but I'm surprised this isn't named  `eval₂_ring_hom_of_commutes`.

Reply:
Yeah, I thought that was weird.

Reply:
So let's take this opportunity to fix the name. I think `eval2_ring_hom_of_commutes` is still awefully long. I'm inclined to just call it `eval\2'` or maybe `eval\2_hom'`. What do you think @awainverse @semorrison ?

Reply:
Well, the name should be similar to `eval₂_ring_hom`, so I'm in favor of either `eval₂_ring_hom_of_commutes` or `eval₂_ring_hom'`

Reply:
Or maybe they should be called `eval₂_ring_hom` and `eval₂_comm_ring_hom`?

Reply:
It would be good to shorten things like `ker_eval₂_ring_hom_noncomm_unit_polynomial`

Reply:
I would suggest going with `eval\2_ring_hom'` for now. I still want to see a refactor of `eval\2` so that it is a ring hom directly, but that's future work. (I'm too busy atm)

Reply:
I've gone with that for now.

Reply:
Ok. I would have prefered `_of_commutes`, but I like long names.

I've thought a few times about making more of the homomorphisms about polynomials into `ring_hom`s from the beginning, but there are good reasons not to...

Reply:
If I had my way, I'd probably call them `comm_ring_eval` and `ring_eval` or even `creval` and `reval`...